src/snowflake/snowpark/functions.py

```diff
@@ -838,10 +976,21 @@ def create_map(*cols: Union[ColumnOrName, Iterable[ColumnOrName]]) -> Column:
             f"The 'create_map' function requires an even number of parameters but the actual number is {len(cols)}"
         )
 
-    return object_construct_keep_null(*cols)
+    # To make Ast deterministic, sort set and convert to tuple.
+    if isinstance(cols, set):
+        cols = tuple(sorted(list(cols)))
```

Comment:
do not do this. This will break the behavior

Reply:
we prob. can fix this by addressing the type hint first, that would then help to clarify behavior/conversions etc.

Reply:
remove with the check added.